_integration-tests/utils/agent/agent.go

```diff
@@ -0,0 +1,188 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package agent
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/beevik/guid"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+type MockAgent struct {
+	virtualEnv     string
+	process        *exec.Cmd
+	processCancel  context.CancelFunc
+	currentSession atomic.Pointer[mockSession]
+	port           int
+}
+
+type mockSession struct {
+	agent *MockAgent
+	token *guid.Guid
+}
+
+func New(t *testing.T) (agent MockAgent, err error) {
```

Comment:
This is not necessary to use named return here because there is no `defer` code so please don't use it as it makes the flow very weird as always

Reply:
I don't see what difference it makes, but okay :)